Packages/com.unity.inputsystem/InputSystem/Devices/NoiseFilter.cs

```diff
@@ -0,0 +1,246 @@
+
+            NoiseFilter filter = new NoiseFilter();
+            FilteredElement* elementsToAdd = stackalloc FilteredElement[device.allControls.Count];
+            int elementCount = 0;
+            ReadOnlyArray<InputControl> controls = device.allControls;
+            for (int i = 0; i < controls.Count; i++)
+            {
+                InputControl control = controls[i];
+                if (control.noisy)
+                {
+                    FilteredElement newElement;
+                    newElement.controlIndex = i;
+                    newElement.type = EElementType.TypeFull;
+                    InputStateBlock stateblock = control.stateBlock;
+                    elementsToAdd[elementCount++] = newElement;
+                }
+                else
+                {
+                    InputControl<float> controlAsFloat = control as InputControl<float>;
+                    if (controlAsFloat != null && controlAsFloat.processors != null)
+                    {
+                        if (controlAsFloat.processors != null)
+                        {
+                            FilteredElement newElement;
+                            newElement.controlIndex = i;
+                            newElement.type = EElementType.TypeFloat;
+                            InputStateBlock stateblock = control.stateBlock;
+                            elementsToAdd[elementCount++] = newElement;
+                        }
+                    }
+                    else
+                    {
+                        InputControl<Vector2> controlAsVec2 = control as InputControl<Vector2>;
+                        if (controlAsVec2 != null && controlAsVec2.processors != null)
+                        {
+                            FilteredElement newElement;
+                            newElement.controlIndex = i;
+                            newElement.type = EElementType.TypeVec2;
+                            InputStateBlock stateblock = control.stateBlock;
+                            elementsToAdd[elementCount++] = newElement;
+                        }
+                    }
+                }
+            }
+
+            filter.elements = new FilteredElement[elementCount];
+            for (int j = 0; j < elementCount; j++)
+            {
+                filter.elements[j] = elementsToAdd[j];
+            }
+
+            return filter;
+        }
+
+        /// <summary>
+        /// The list of elements to be checked for.  Each element represents a single InputControl.
+        /// </summary>
+        public FilteredElement[] elements;
+
+        /// <summary>
+        /// Called when the NoiseFilter gets applied to a device, calls down to any individual FilteredElements that need to do any work.
+        /// </summary>
+        /// <param name="device">The device you want to apply filtering to.</param>
+        internal void Apply(InputDevice device)
```

Comment:
This one I'm not so sure.  
I want to keep it open, in case future changes to the NoiseFilter need to do something other than just set the bitmask on apply.